src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -2055,6 +2075,23 @@ private void getFieldInfo(ref CORINFO_RESOLVED_TOKEN pResolvedToken, CORINFO_MET
                         helperId = ReadyToRunHelperId.GetNonGCStaticBase;
                     }
 
+#if READYTORUN
+                    if (!_compilation.NodeFactory.CompilationModuleGroup.ContainsType(field.OwningType))
+                    {
+                        // Static fields outside of the version bubble need to be accessed using the ENCODE_FIELD_ADDRESS
+                        // helper in accordance with ZapInfo::getFieldInfo in CoreCLR.
+                        pResult->fieldLookup = CreateConstLookupToSymbol(_compilation.SymbolNodeFactory.FieldAddress(field, _signatureContext));
+
+                        pResult->helper = CorInfoHelpFunc.CORINFO_HELP_READYTORUN_STATIC_BASE;
+
+                        fieldFlags &= ~CORINFO_FIELD_FLAGS.CORINFO_FLG_FIELD_STATIC_IN_HEAP; // The dynamic helper takes care of the unboxing
+                        
+                        // Skip emission of "normal" lookup in the conditional block below and fall through to the final chores
```

Comment:
Setting the helper to Invalid and this comment can be avoided by adding an `else` before the `#endif`.

Reply:
Fixed.